src/components/label/label.wp.scss

```diff
@@ -61,8 +61,8 @@ html[dir="rtl"] .label-wp[floating] {
   transform: translate3d(0, 0, 0) scale(.8);
 }
 
-html[dir="rtl"] .input-has-focus .label-wp[floating],
-html[dir="rtl"] .input-has-value .label-wp[floating] {
+[dir="rtl"] .input-has-focus .label-wp[floating],
```

Comment:
because I've added a `transform: translate3d` with more specific reference then a class (an attribute) it will replace the original LTR (`.input-has-focus or .input-has-value`), with default one so the label won't float on RTL direction, therefore I've added the same thing with attribute specific (`dir="rtl"`) ... it's hackish, I know, but couldn't find a better suitable easy way.